nautobot/utilities/utils.py

```diff
@@ -530,3 +530,26 @@ def get_table_for_model(model):
 
 # Setup UtilizationData named tuple for use by multiple methods
 UtilizationData = namedtuple("UtilizationData", ["numerator", "denominator"])
+
+
+def versioned_serializer_selector(obj, legacy_serializer, serializer):
```

Comment:
Love this cleanup but the version detection needs to be flexible. Maybe this can take a dictionary for the "introduced in" version, or at least as an argument. Because for the new serializer for Interface status, that needs to be in 1.4.

Reply:
I'm thinking a dictionary like this:
```py
serializer_choices={
  "1.3": legacy_serializer=serializers.InterfaceSerializerVersion12,
  "1.4": serializer=super().get_serializer_class(), 
}
```

Where the key is the "version introduced in". (one footnote is that our current InterfaceSerializer today should be Version12, but the new one being implemented here is 1.4 since that's the version it's being introduced in)

We might need to workshop this a bit to help determine what the default is, etc. I love the code reduction but it's not going to scale as we progress with more changes.